nemo_skills/inference/genselect.py

```diff
@@ -0,0 +1,199 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import logging
+import re
+import sys
+from dataclasses import field
+from enum import Enum
+from pathlib import Path
+from copy import deepcopy
+import random
+
+import hydra
+import typer
+from typing import Any
+from tqdm import tqdm
+from os import path, makedirs
+from nemo_skills.inference.server.code_execution_model import server_params
+from nemo_skills.inference.generate import InferenceConfig, GenerationTask
+from nemo_skills.utils import get_help_message, nested_dataclass, setup_logging
+
+LOG = logging.getLogger(__file__)
+
+
+@nested_dataclass(kw_only=True)
+class GenSelectConfig:
+    """Genselect parameters."""
+
+    input_dir: str  # Directory where the original predictions are saved
+    output_dir: str  # Where to save the intermediate outputs and final predictions
+    # Inference server configuration {server_params}
+    server: dict = field(default_factory=dict)
+    # Prompt configuration - path to yaml files
+    prompt_template: str | None = None  # not required for OpenAI server
+    prompt_config: str = "openmath/genselect"  # GenSelect template
+
+    inference: InferenceConfig = field(default_factory=InferenceConfig)  # LLM call parameters
+
+    # Can specify one of the existing datasets.
+    dataset: str | None = None
+    split: str | None = None  # Generally one of train/test, but can be anything since it's used as part of a file name
+
+    batch_size: int = 128
+    max_samples: int = -1  # If > 0, will stop after generating this many samples. Useful for debugging
+    skip_filled: bool = False  # If True, will skip the generations that are already in the output file
+
+    max_concurrent_requests: int = 1024  # Maximum number of concurrent requests to the server for the async loop
+    # chunk the dataset into equal sized parts and index into them
+    num_chunks: int | None = None  # if specified, will split the data into chunks and only generate for one chunk
+    chunk_id: int | None = None  # if specified, will index the specified chunk only
+
+    generation_key: str = "gen_rm_comparison"
```

Comment:
```suggestion
    generation_key: str = "genselect_comparison"
```
I guess might need some updates in other scripts for this? Or gen_rm_comparison = genselect?